code/controllers/subsystem/wardrobe.dm

```diff
@@ -236,21 +263,34 @@ SUBSYSTEM_DEF(wardrobe)
 		stock_info[WARDROBE_STOCK_CALL_REMOVAL] = master_info[WARDROBE_CACHE_CALL_REMOVAL]
 		preloaded_stock[object_type] = stock_info
 
+	if(object.loc != null)
+		object.moveToNullspace()
 	var/datum/callback/do_on_insert = stock_info[WARDROBE_STOCK_CALL_INSERT]
 	if(do_on_insert)
-		do_on_insert.object = object
-		do_on_insert.Invoke()
-		do_on_insert.object = null
+		do_on_insert.FleetingInvoke(object)
 
-	object.moveToNullspace()
 	stock_info[WARDROBE_STOCK_CONTENTS] += object
+	return TRUE
 
-/datum/controller/subsystem/wardrobe/proc/provide_type(datum/requested_type, atom/movable/location)
+/// Returns an object of requested_type at location, alongside a set of of lists of a defined key and arguments to call on the new object
+/datum/controller/subsystem/wardrobe/proc/provide(datum/requested_type, atom/movable/location, ...)
```

Comment:
I think this could use some beefed up documentation given how much user facing code it's being stuck in. 

"When should I be using this" as an example.

Reply:
yeah, u right. will get to it tomorrow